packages/eas-cli/src/project/remoteVersionSource.ts

```diff
@@ -98,3 +123,84 @@ export function getBuildVersionName(platform: Platform): string {
     return 'buildNumber';
   }
 }
+
+export async function ensureAppVersionSourceIsSetAsync(
+  easJsonAccessor: EasJsonAccessor,
+  easJsonCliConfig: EasJson['cli'] | undefined,
+  nonInteractive: boolean
+): Promise<EasJson['cli'] | undefined> {
+  if (nonInteractive) {
+    Log.error('Cannot select "appVersionSource" in non-interactive mode.');
+    Log.error(
+      'Run the command again in interactive mode or configure the "appVersionSource" manually.'
+    );
+    Log.error(
+      learnMore('https://docs.expo.dev/build-reference/app-versions/', {
+        learnMoreMessage: 'See the docs to learn more.',
+        dim: false,
+      })
+    );
+    Errors.error('"appVersionSource" must be configured in non-interactive mode', { exit: 1 });
+  }
+
+  Log.log(
+    `Please select your app version source. With "local" the build android.versionCode / ios.buildNumber needs to be incremented manually in app.json / app.config.js, and with "remote" this value is stored in EAS and will be incremented automatically with each build. Until now, this project has been using the "local" version source (which was previously the default). App version source can be set for you automatically, or you can configure it yourself - if you wish to use the default "remote" version source add ${chalk.bold(
+      '{"cli": { "appVersionSource": "remote" }}'
+    )} to your eas.json or if you want to use the "local" version source add ${chalk.bold(
+      '{"cli": { "appVersionSource": "local" }}'
+    )} to your eas.json.`
+  );
```

Comment:
So what would you do if they don't have autoIncrement:true / have runtimeVersion.policy: nativeVersion / have autoIncrement: version ?
Don't prompt them and just set it to local, plus maybe display a message that it has been done?

Reply:
> if they don't have autoIncrement:true

I think we can just print a warning that this field will be required in the future, but not stop them from doing what they do to select the version source in a prompt? Does it make sense? Most of our users don't know what the version source is, so it would be confusing to them to be forced to understand it and set it, while they never relied on it.

>have runtimeVersion.policy: nativeVersion / have autoIncrement: version

Set the source  to `local`

@kadikraman what do you think?

Reply:
Hm, I'm wondering how best to approach it - the `appVersionSource` setting is profile-agnostic, while autoIncrement is not. If the user runs a non-production build then almost certainly `autoIncrement` is not set, but it could be for a `production` profile 🤔